app/models/ChangeRequest.ts

```diff
@@ -0,0 +1,10 @@
+import { ScheduleDay } from './Meta';
+
+export type ChangeRequest = any
+export type ChangeRequestParams = ScheduleDayChangeRequestParams
+
+interface ScheduleDayChangeRequestParams {
+  schedule_id: number
+  type: 'schedule_days'
+  change_request: Partial<ScheduleDay>
```

Comment:
Change requests are the only case I feel it is worth doing this, since one change request accepts wildly different params. My assumption being ChangeRequest is a union type of any one of the allowed params types.

I feel like anything else will just get complex and confuse newbies to the codebase.

Reply:
Right, I think what I was saying would only apply to ChangeRequests, since it's pretty much the only way to create objects. I'm not sure if we can get around the complexity of having separate read v.s. write types, since the API is just different, unless if we wanted to hoist all of that into the frontend app.

But agreed that we can revise this in the future.